sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/PartitionReceiverLiveTests.cs

```diff
@@ -2246,17 +2278,21 @@ await using (var producer = new EventHubProducerClient(connectionString))
                     {
                         batchHasEvents = true;
 
-                        var eventId = eventData.Properties[EventGenerator.IdPropertyName].ToString();
+                        result.RemainingEvents?.Remove(eventData.MessageId);
+                        result.Events.TryAdd(eventData.MessageId, eventData);
+
+                        // If there's a callback registered per-tick, invoke it and respect its
+                        // decision on whether iteration should continue.
 
-                        if ((result.Events.TryAdd(eventId, eventData)) && (result.Events.Count >= expectedEventCount))
+                        if ((iterationCallback != null) && (!(await iterationCallback(result).ConfigureAwait(false))))
```

Comment:
nit: Based on its usage, it seems like `iterationCallback` could be renamed to `shouldReadNextEventBatchCallback` and this whole block could be simplified to:
```suggestion
                        shouldReadNextEventBatch = result.RemainingEvents?.Count switch {
                            not null => false,
                            null when shouldReadNextEventBatchCallback is not null => shouldReadNextEventBatchCallback(result).ConfigureAwait(false),
                            _ => true
                        };
                      
```

Reply:
I think that changes the logic; reads are done when the count is 0 and the callback should always run if provided, regardless of whether there are events left to read, as there are side-effects that take place in some scenarios (for example, setting a completion source).   I probably could create a compound switch clause to cover things, but I'm going to keep the simple form.